### PR TITLE
[FIX] Adds support for adding/removing numbers and list accesses to/from lists (#3293)

### PR DESCRIPTION
**Pull request by jakeklaris**
[FIX] added support for adding/removing numbers and list accesses to/from lists (#3293)

**Description**

Adds support to levels 3+ for adding integers or floats to lists, and removing integers of floats from lists. Also fixes the issue where accessing a list item with square bracket indexing gets changed into a literal string when adding or removing from a list (ex. add list1[2] to list2, appends 'list1[2]' as a string to list2)

**Fixes Issue #3293**

**How to test**

Adds a test to test_level_03.py for checking that a number can be added to a list
Adds tests to test_level_16.py for checking that square bracket list indexing correctly gets added and removed from a list

### CODE REVIEW

**Comment by Felienne** [comment]
Welcome to Hedy @jakeklaris and well done on picking up a language bug as your first PR! We can simplify the code a bit but it does seem to work (leaving a few details comments now)

File: hedy.py [hunk 1]
```diff
@@ -1469,12 +1469,27 @@ def list_access(self, meta, args):
...
+            value = before_index + '-1' + ']' + after_index # account for 1-based indexing
+        else:
+            value = self.process_variable(args[0], meta.line) 
+
         list_var = args[1]
         return f"{list_var}.append({value})"
 
     def remove(self, meta, args):
-        value = self.process_variable(args[0], meta.line)
+        if args[0].isnumeric(): # is int/float
```
Comment (Felienne) [inline hedy.py:1485]: Would be good to repeat the comment from add here (future readers will not always think to read add when looking for more info on remove)
```suggestion
      # only process arg if arg is a string, else keep as is (ie. don't change 5 into '5', my_list[1] into 'my_list[1]')
        if args[0].isnumeric(): # is int/float
```

File: hedy.py [hunk 2]
```diff
@@ -1469,12 +1469,27 @@ def list_access(self, meta, args):
...
+            value = args[0]
+        elif ('[' in args[0] and ']' in args[0]): # is list indexing
+            before_index, after_index = args[0].split(']', 1)
+            value = before_index + '-1' + ']' + after_index 
+        else:
+            value = self.process_variable(args[0], meta.line)
```
Comment (Felienne) [inline hedy.py:1491]: These lines are exactly the same as for add, can we make a function for this and call it from both places (a bit like arithmetic operators do in level 6 by all calling `process_calculation()`)

**Review by Felienne — CHANGES_REQUESTED** [review]
These are all supersmall style tips, the mechanic seems to work just fine!

**Comment by jakeklaris** [comment]
> These are all supersmall style tips, the mechanic seems to work just fine!

Great! Just updated the code to account for the style tips. Let me know what you think.

**Review by Felienne — APPROVED** [review]
Thanks for the fixes, and congrats on getting your first PR merged @jakeklaris!

**Pull request merged**